File: SPRING/Hospital_Management_system_angular/src/app/services/admission.service.ts
import { Injectable } from '@angular/core';
import { environment } from '../../environments/environment';
import { HttpClient } from '@angular/common/http';
import { WardModel } from '../models/ward.model';
import { BedModel } from '../models/bed.model';
import { AdmissionRequest } from '../models/admission.model';
import { AdmissionResponse } from '../models/admission-response.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root',
})
export class AdmissionService {


  // api = environment.apiUrl + "infrastructure";

  // constructor(private http: HttpClient) { }

  // getAllWards() {
  //   return this.http.get<WardModel[]>(this.api + "/wards");
  // }

  // getBedsByWard(id: number) {
  //   return this.http.get<BedModel[]>(`${this.api}/wards/${id}/beds`);
  // }




  private api = environment.apiUrl + "admissions";

  constructor(private http: HttpClient) { }

  admit(data: AdmissionRequest): Observable<AdmissionResponse> {

    return this.http.post<AdmissionResponse>(
      `${this.api}/admit`,
      data
    );

  }

  discharge(id: number): Observable<AdmissionResponse> {

    return this.http.put<AdmissionResponse>(
      `${this.api}/discharge/${id}`,
      {}
    );

  }


  getAll(): Observable<AdmissionResponse[]> {

    return this.http.get<AdmissionResponse[]>(this.api);

  }

  getById(id: number): Observable<AdmissionResponse> {

    return this.http.get<AdmissionResponse>(`${this.api}/${id}`);

  }

}

File: Flutter_Hospital_Management/lib/services/admission_service.dart
import 'package:dio/dio.dart';
import 'package:flutter_hospital_management/models/admission.dart';

class AdmissionService {
  final Dio dio;

  AdmissionService(this.dio);

  Future<List<Admission>> getAll() async {
    final res = await dio.get('/admissions');
    return (res.data as List)
        .map((e) => Admission.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  Future<List<Admission>> getActive() async {
    final res = await dio.get('/admissions/active');
    return (res.data as List)
        .map((e) => Admission.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  Future<Admission> admit(AdmissionRequest request) async {
    final res = await dio.post('/admissions/admit', data: request.toJson());
    return Admission.fromJson(res.data as Map<String, dynamic>);
  }

  Future<Admission> discharge(int id) async {
    final res = await dio.put('/admissions/discharge/$id');
    return Admission.fromJson(res.data as Map<String, dynamic>);
  }
}
